sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -0,0 +1,243 @@
+          positive class (index 1) is always used.
+        - In a multioutput setting, specifies the task for which the PDPs
+          should be computed
+        Ignored in binary classification or classical regression settings.
+
+    pdp_lim : dict
+        Global min and max average predictions. `pdp_lim[1]` is the global min
+        and max for single partial dependence curves. `pdp_lim[2]` is the
+        global min and max for two-way partial dependence curves.
+
+    deciles : dict
+        Deciles for feature indicies in ``features``.
+
+    Attributes
+    ----------
+    bounding_ax_ : matplotlib Axes or None
+        If `ax` is an axes or None, the `bounding_ax_` is the axes where the
+        grid of partial dependence plots are drawn. If `ax` is  list of axes,
+        `bounding_ax_` is None.
+
+    axes_ : dict of matplotlib Axes
+        If `ax` is an axes or None, `axes_[i, j]` is the axes on the ith row
+        and jth column. If `ax` is a list of axes, `axes_[i]` is the ith item
+        in `ax`.
+
+    lines_ : matplotlib Artists
+        If `ax` is an axes or None, `line_[i, j]` is the partial dependence
+        curve on the ith row and jth column. If `ax` is a list of axes,
+        `lines_[i]` is the partial dependence curve corresponding to the ith
+        item in `ax`. Only plots with a single feature will be populated in
+        `lines_`. Contour plots are stored in `contours_`.
+
+    contours_ : matplotlib Artists
+        If `ax` is an axes or None, `contours_[i, j]` is the partial dependence
+        plot on the ith row and jth column. If `ax` is a list of axes,
+        `contours_[i]` is the partial dependence plot corresponding to the ith
+        item in `ax`. Only plots with a two feature will be populated in
+        `contours_`. Line plots are stored in `lines_`.
+
+    figure_ : matplotlib Figure
+        Figure containing partial dependence plots.
+    """
+    def __init__(self, pd_results, features, feature_names, target_idx,
+                 pdp_lim, deciles):
+        self.pd_results = pd_results
+        self.features = features
+        self.feature_names = feature_names
+        self.target_idx = target_idx
+        self.pdp_lim = pdp_lim
+        self.deciles = deciles
+
+    def plot(self, ax=None, n_cols=3, line_kw=None, contour_kw=None, fig=None):
+        """Plot partial dependence plots.
+
+        Parameters
+        ----------
+        ax : Matplotlib axes, list of Matplotlib axes or None, default=None
+            Axes to plot the partial dependence curves. If a single axes is
+            given, it is treated as a bounding axes and a grid of partial
+            depdendence plots will be drawn on that top of it. If a list of
+            axes are passed, the partial dependence plots will be drawn on
+            those axes. By default, a single bounding axes is created and
+            treated as the single axes case.
+
+        n_cols : int, default=3
```

Comment:
Honestly, I hate the way this is currently done and we should be doing what dabl is doing or https://github.com/matplotlib/grid-strategy is doing but that's for another day lol.